nixos/modules/security/pam.nix

```diff
@@ -1472,6 +1472,15 @@ in
       }
     ];
 
+    warnings = optional
+      (with lib; with config.security.pam.sshAgentAuth; enable -> any (hasPrefix "%h") authorizedKeysFiles)
+      ''config.security.pam.sshAgentAuth.authorizedKeysFiles contains files in the user's home directory.
+
+        This is an insecure configuration, as a malicious actor who gained code-execution as the user
+        can then edit the trusted authorized_keys file and thus bypass the ssh-agent-based authentication.
+        See https://github.com/NixOS/nixpkgs/issues/31611
```

Comment:
This explanation would be helpful in the `authorizedKeysFile` description, since users could end up using a user-writable path that doesn't trigger this warning.